src/Core/OrganizationFeatures/Groups/CreateGroupCommand.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.Data;
+using Bit.Core.OrganizationFeatures.Groups.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.OrganizationFeatures.Groups;
+
+public class CreateGroupCommand : ICreateGroupCommand
+{
+    private readonly IEventService _eventService;
+    private readonly IGroupRepository _groupRepository;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IReferenceEventService _referenceEventService;
+
+    public CreateGroupCommand(
+        IEventService eventService,
+        IGroupRepository groupRepository,
+        IOrganizationRepository organizationRepository,
+        IReferenceEventService referenceEventService)
+    {
+        _eventService = eventService;
+        _groupRepository = groupRepository;
+        _organizationRepository = organizationRepository;
+        _referenceEventService = referenceEventService;
+    }
+
+    public async Task CreateGroupAsync(Group group,
+        IEnumerable<SelectionReadOnly> collections = null)
+    {
+        await GroupRepositoryCreateGroupAsync(group, collections);
+        await _eventService.LogGroupEventAsync(group, Enums.EventType.Group_Created);
+    }
+
+    public async Task CreateGroupAsync(Group group, EventSystemUser systemUser,
+        IEnumerable<SelectionReadOnly> collections = null)
+    {
+        await GroupRepositoryCreateGroupAsync(group, collections);
+        await _eventService.LogGroupEventAsync(group, Enums.EventType.Group_Created, systemUser);
+    }
+
+    private async Task GroupRepositoryCreateGroupAsync(Group group, IEnumerable<SelectionReadOnly> collections = null)
+    {
+        var organization = await _organizationRepository.GetByIdAsync(group.OrganizationId);
+        if (organization == null)
+        {
+            throw new BadRequestException("Organization not found");
+        }
```

Comment:
We can't have database reads in a command. I can see why you changed this (to remove the extraneous parameter), but it's out of place here.

On the other hand, requiring the `Organization` object to be passed in just to check its `Id` (when we're not otherwise doing anything with it) seems a bit pointless. I'm inclined to remove it altogether, or move it up to the controller level.

Could you please check with @MGibson1?

Reply:
yup, I’d move that up to the controller to grab the org, create another method on the command class (`Validate` or some such) and use that method to check the org exists and can use groups

Reply:
Right, I missed the `CanUseGroups` check. Makes sense, ty

Reply:
Thank you both for your suggestions. I've extracted the database read from the command to the respective controllers and added unit tests around them.